src/preflight/handler.js

```diff
@@ -0,0 +1,225 @@
+    options: {
+      enableAuthentication: true,
+    },
+  };
+}
+
+export const preflightAudit = async (context) => {
+  const {
+    site, job, s3Client, log,
+  } = context;
+
+  const jobMetadata = job.getMetadata();
+  const { urls } = jobMetadata.payload;
+
+  if (!validPages(urls)) {
+    throw new Error(`[preflight-audit] site: ${site.getId()}. Invalid pages provided`);
+  }
+
+  if (job.getStatus() !== AsyncJob.Status.IN_PROGRESS) {
+    throw new Error(`[preflight-audit] site: ${site.getId()}. Job not in progress for jobId: ${job.getId()}. Status: ${job.getStatus()}`);
+  }
+
+  const result = {
+    audits: [],
+  };
+
+  // canonical
+  result.audits.push({
+    name: 'canonical',
+    type: 'seo',
+    opportunities: [],
+  });
+
+  const urlsSet = new Set(urls.map((page) => {
+    const pathname = new URL(page.url).pathname.replace(/\/$/, '');
+    return `scrapes/${site.getId()}${pathname}/scrape.json`;
+  }));
+
+  const bucketName = context.env.S3_SCRAPER_BUCKET_NAME;
+  const prefix = `scrapes/${site.getId()}/`;
+  const scrapedObjectKeys = await getObjectKeysUsingPrefix(s3Client, bucketName, prefix, log);
+  const scrapedObjects = await Promise.all(
+    scrapedObjectKeys
+      .filter((key) => urlsSet.has(key))
+      .map(async (key) => ({
+        url: key,
+        data: await getObjectFromKey(s3Client, bucketName, key, log),
+      })),
+  );
+
+  // internal-links
+  result.audits.push({
+    name: 'internal-links',
+    type: 'seo',
+    opportunities: [],
+  });
+
+  const pageAuthToken = await retrievePageAuthentication(site, context);
+
+  const brokenLinks = await runInternalLinkChecks(scrapedObjects, pageAuthToken, context);
+  if (!isNonEmptyArray(brokenLinks)) {
+    const baseUrl = site.getBaseURL();
+    const suggestionData = await generateSuggestionData(baseUrl, brokenLinks, context, site);
+    for (const url of urls) {
+      const { url: pageUrl } = url;
+      const suggestions = suggestionData[pageUrl];
+      if (suggestions) {
+        result.audits[1].opportunities.push({
+          url: pageUrl,
+          suggestions,
+        });
+      }
+    }
+  }
+
+  log.info('INTERNAL LINKS CHECKS', JSON.stringify(result, null, 2));
+
+  // metatags
+  result.audits.push({
+    name: 'metatags',
+    type: 'seo',
+    opportunities: [],
+  });
+
+  const {
+    seoChecks, detectedTags, extractedTags,
+  } = await metatagsAutoDetect(site, urlsSet, context);
+
+  const allTags = {
+    detectedTags,
+    healthyTags: seoChecks.getFewHealthyTags(),
+    extractedTags,
+  };
+
+  if (Object.keys(detectedTags).length > 0) {
+    const updatedDetectedTags = await metatagsAutoSuggest(allTags, context, site);
+    for (const url of urls) {
+      const { url: pageUrl } = url;
+      const path = new URL(pageUrl).pathname.replace(/\/$/, '');
+      const tags = updatedDetectedTags[path];
+      if (tags) {
+        result.audits[2].opportunities.push({
+          ...tags,
+        });
+      }
+    }
+  }
```

Comment:
allTags only needed inside if statement and simplify:
```suggestion
if (Object.keys(detectedTags).length > 0) {
  const allTags = {
    detectedTags,
    healthyTags: seoChecks.getFewHealthyTags(),
    extractedTags,
  };

  const updatedDetectedTags = await metatagsAutoSuggest(allTags, context, site);

  for (const { url } of urls) {
    const path = new URL(url).pathname.replace(/\/$/, '');
    const tags = updatedDetectedTags[path];
    if (tags) {
      result.audits[2].opportunities.push(tags);
    }
  }
}
```